### PR TITLE
Move QMCLinearOptimizeBatched to QMCFixedSampleLinearOptimizeBatched

### PR DESCRIPTION
**Pull request by markdewing**
Move remaining functionality from QMCLinearOptimizeBatched to QMCFixedSampleLinearOptimizeBatched and remove QMCLinearOptimizeBatched.
 
Also removed commented-out code from generalized eigenvalue solvers

Future: The "FixedSample" part of the name is no longer necessary. A future PR will rename the class to QMCLinearOptimizeBatched. It was easier step-wise to move everything to the more specific QMCFixedSampleLinearOptimizeBatched class first.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- N/A This PR adds tests to cover any new code, or to catch a bug that is being fixed
- N/A Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
Changes LGTM.

About the future naming, removing FixedSample makes batched class doesn't look like the one wihtout batch. I mean QMCFixedSampleLinearOptimize correspond to QMCLinearOptimizeBatched It is a bit confusing.
How easy it is to apply the same class fusion to QMCFixedSampleLinearOptimize?

**Comment by markdewing** [comment]
I would like to avoid doing too much with the non-batched classes.  I assume they will go away eventually.   Adding some comments about the class correspondence should be sufficient to reduce confusion.   I can also put off renaming QMCFIxedSampleLinearOptimizeBatched since it's not critical.

File: src/QMCDrivers/WFOpt/QMCFixedSampleLinearOptimizeBatched.cpp [hunk 1]
```diff
@@ -1493,4 +1600,154 @@ bool QMCFixedSampleLinearOptimizeBatched::hybrid_run()
 }
 #endif
 
+QMCFixedSampleLinearOptimizeBatched::RealType QMCFixedSampleLinearOptimizeBatched::getLowestEigenvector(
```
Comment (PDoakORNL) [inline src/QMCDrivers/WFOpt/QMCFixedSampleLinearOptimizeBatched.cpp:1603]: Unit test?  Doesn't belong in this PR but it should be in one.

**Review by PDoakORNL — APPROVED** [review]
Other than maybe revealing a missing unit test this refactor looks good to me and removes unecessary inheritance.

**Comment by ye-luo** [comment]
> I would like to avoid doing too much with the non-batched classes. I assume they will go away eventually. Adding some comments about the class correspondence should be sufficient to reduce confusion. I can also put off renaming QMCFIxedSampleLinearOptimizeBatched since it's not critical.

I vote for just not doing the renaming. The only downside is long class name but it is just minor issue.

**Comment by ye-luo** [comment]
Start testing in-house
